controllers/managedocs_controller.go

```diff
@@ -126,11 +188,51 @@ func (r *ManagedOCSReconciler) setDesiredStorageCluster(
 	if reconcileStrategy == v1.ReconcileStrategyStrict {
 		// Get an instance of the desired state
 		desired := utils.ObjectFromTemplate(templates.StorageClusterTemplate, r.Scheme).(*ocsv1.StorageCluster)
-
+		r.updateStorageClusterWithParams(desired, cs)
 		// Override storage cluster spec with desired spec from the template.
 		// We do not replace meta or status on purpose
 		sc.Spec = desired.Spec
+		r.Log.Info("StorageCluster", "desiredSpec", desired.Spec)
+		r.Log.Info("StorageCluster", "Current spec", sc.Spec)
 	}
 
 	return nil
 }
+
+func (r *ManagedOCSReconciler) updateStorageClusterWithParams(
+	sc *ocsv1.StorageCluster,
+	cs *corev1.Secret) error {
+
+	if cs == nil {
+		return nil
+	}
+	r.Log.Info("Updating storagecluster")
+
+	sdata := cs.Data
+
+	for key, element := range sdata {
+		r.Log.Info("Secret keys:", "key", key, "value", string(element))
+		PersistentVolume := &sc.Spec.StorageDeviceSets[0].DataPVCTemplate
+		if isValidStorageClusterParam(key, string(sdata[key])) {
+			PersistentVolume.Spec.Resources = corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					"storage": resource.MustParse(string(sdata[key])),
+				},
+			}
+		}
+		r.Log.Info("Post updation Persistent Volume ", "PV", PersistentVolume)
```

Comment:
Why do we need this generic code? can we just do a simple read of predefined keys from the secret? 
It will be easier to read, understand and even handle special cases (for future params) more easily (for example 2 params that depend on each other)

```
storageClassSize := cs.data[storageClassSizeKey]
if storageClassSize != "1TiB" || storageClassSize != "4TiB" {
    return r.Log.Ferrorf("Invalid storage class size, %s = %s", storageClassSizeKey, storageClassSize)
}
PersistentVolume := &sc.Spec.StorageDeviceSets[0].DataPVCTemplate
PersistentVolume.Spec.Resources["storage"] = resource.MustParse(storageClassSize)
```

For that matter, I suggest that this whole code section should be folded back into the `setDesiredStorageCluster` which is the place to update the storage cluster

Reply:
This was part of the code I was using to understand the way the secret worked as part of the draft code.

Reply:
Ack